model/lifecycle.go

```diff
@@ -1426,23 +1421,31 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 	taskIds := []string{}
 	for _, b := range builds {
 		// Find the set of task names that already exist for the given build
-		tasksInBuild, err := task.Find(task.ByBuildId(b.Id).WithFields(task.DisplayNameKey))
+		tasksInBuild, err := task.Find(task.ByBuildId(b.Id))
 		if err != nil {
 			return nil, err
 		}
 		// build an index to keep track of which tasks already exist
-		existingTasksIndex := map[string]bool{}
+		existingTasksIndex := map[string]task.Task{}
```

Comment:
If you use withfields, you could keep as many maps as there are interesting fields (seems like 2, DisplayName and Activated).

Reply:
The reason I didn't do this is because SetActiveState also looks at DispatchTime, Status, OverrideDependencies, Dependencies, etc, so it seems like too many fields then.